### PR TITLE
Move to using proto3-wire from upstream

### PR DESCRIPTION
**Pull request by neil-da**
One less dependency. If this passes CI I intend to fork proto3-wire (so we have a backup), ask for a release on the upstream repo (so we can eliminate the fork entirely) and delete it from `nix` (less code).

### CODE REVIEW

**Comment by neil-da** [comment]
Two issues:

* We've made some optimisations. Upstream have made some optimisations. We need to see what we can move upstream, what is duplicate etc.
* There is a semantic difference when I switch to upstream. I'm still tracking down why.

**Comment by neil-da** [comment]
This PR is very much in draft. Before merging I need to:

- [ ] Merge my changes into awakesecurity upstream
- [ ] Fork the awakesecurity stuff into digital-asset so we have a backup of them
- [ ] Order dependencies, delete empty deps etc
- [ ] Delete the things in Nix that are no longer required

File: 3rdparty/haskell/BUILD.old-time [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# vim: ft=python
+# By default 'old-time' fails to build on Windows and is very much inhermetic on
+# Linux/Darwin. The reasons are described here: https://github.com/FormationAI/hazel/issues/80
+#
+# In order to work around this we use a cabal-like 'package.bzl' generated on
+# Linux ('network-package.bzl') and tweak it depending on the platform.
```
Comment (cocreature) [inline 3rdparty/haskell/BUILD.old-time]:
```suggestion
# Linux ('old-time-package.bzl') and tweak it depending on the platform.
```

File: 3rdparty/haskell/BUILD.old-time [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# * Some modules are not built on Windows (Posix-specific)
+# * -Dmingw32_HOST_OS and -Dx86_64_HOST_ARCH are set on Windows
+# * the 'unix' dependency is dropped on Windows
+# * on all platform we introduce a C dependency on the generated network
```
Comment (cocreature) [inline 3rdparty/haskell/BUILD.old-time]:
```suggestion
# * on all platform we introduce a C dependency on the generated old-time
```

File: 3rdparty/haskell/BUILD.old-time (hunk 1 above)
Comment (aherrmann-da) [inline 3rdparty/haskell/BUILD.old-time]:
```suggestion
# Linux ('old-time-package.bzl') and tweak it depending on the platform.
```

File: bazel_tools/haskell-optp-response.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+index 20c8720..7fca51c 100644
+--- a/haskell/private/actions/compile.bzl
++++ b/haskell/private/actions/compile.bzl
+@@ -213,7 +213,17 @@ def _compilation_defaults(hs, cc, java, dep_info, plugin_dep_info, srcs, import_
+             set.mutable_insert(import_dirs, idir)
+
+     ghc_args += ["-i{0}".format(d) for d in set.to_list(import_dirs)]
+-    ghc_args += ["-optP" + f for f in cc.cpp_flags]
++
++    # Write the -optP flags to a response because they can be very long on Windows
```
Comment (cocreature) [inline bazel_tools/haskell-optp-response.patch]: Is this whole section only used on Windows? Not entirely obvious from the patch.

Comment (neil-da) [reply bazel_tools/haskell-optp-response.patch]: No, we always use response files for -optP. Doesn't harm on Linux and makes it more consistent.

File: 3rdparty/haskell/BUILD.old-time (hunk 2 above)
Comment (neil-da) [reply 3rdparty/haskell/BUILD.old-time]: I rewrote the comment so it's now accurate

**Pull request merged**